File: pylith/materials/RheologyHeat.py
# =================================================================================================
# This code is part of PyLith, developed through the Computational Infrastructure
# for Geodynamics (https://github.com/geodynamics/pylith).
#
# Copyright (c) 2010-2025, University of California, Davis and the PyLith Development Team.
# All rights reserved.
#
# See https://mit-license.org/ and LICENSE.md and for license information. 
# =================================================================================================

from pylith.utils.PetscComponent import PetscComponent


class RheologyHeat(PetscComponent):
    """
    Abstract base class for bulk rheology for heat conduction.
    """

    import pythia.pyre.inventory

    from spatialdata.spatialdb.SimpleDB import SimpleDB
    auxiliaryFieldDB = pythia.pyre.inventory.facility(
        "db_auxiliary_field", family="spatial_database", factory=SimpleDB)
    auxiliaryFieldDB.meta['tip'] = "Database for physical property parameters."

    def __init__(self, name):
        """Constructor.
        """
        PetscComponent.__init__(self, name, facility="heat_rheology")

    def preinitialize(self, problem):
        """Do pre-initialization setup.
        """
        self._createModuleObj()

    def addAuxiliarySubfields(self, material, problem):
        """Add subfields to auxiliary field.
        """
        pass

    def _createModuleObj(self):
        """Create handle to C++ object.
        """
        raise NotImplementedError("Implement in derived class.")


# End of file

File: libsrc/pylith/materials/RheologyHeat.cc
// =================================================================================================
// This code is part of PyLith, developed through the Computational Infrastructure
// for Geodynamics (https://github.com/geodynamics/pylith).
//
// Copyright (c) 2010-2025, University of California, Davis and the PyLith Development Team.
// All rights reserved.
//
// See https://mit-license.org/ and LICENSE.md and for license information.
// =================================================================================================

#include <portinfo>

#include "pylith/materials/RheologyHeat.hh" // implementation of object methods

#include "pylith/utils/error.hh" // USES PYLITH_METHOD_BEGIN/END
#include "pylith/utils/journals.hh" // USES PYLITH_COMPONENT_*

// ---------------------------------------------------------------------------------------------------------------------
// Default constructor.
pylith::materials::RheologyHeat::RheologyHeat(void) {}


// ---------------------------------------------------------------------------------------------------------------------
// Destructor.
pylith::materials::RheologyHeat::~RheologyHeat(void) {
    deallocate();
} // destructor


// ---------------------------------------------------------------------------------------------------------------------
// Deallocate PETSc and local data structures.
void
pylith::materials::RheologyHeat::deallocate(void) {}


// ---------------------------------------------------------------------------------------------------------------------
// Update kernel constants.
void
pylith::materials::RheologyHeat::updateKernelConstants(pylith::real_array* kernelConstants,
                                                       const PylithReal dt) const {
    PYLITH_METHOD_BEGIN;
    PYLITH_COMPONENT_DEBUG("updateKernelConstants(kernelConstants="<<kernelConstants<<", dt="<<dt<<")");

    assert(kernelConstants);
    // Default is no constants to update.

    PYLITH_METHOD_END;
} // updateKernelConstants


// End of file
